Improves pagination for notifications

valentinyanakiev opened the pull request:
Enhances pagination logic for in-app notifications to ensure correct ordering and stability, especially when using cursor-based pagination.

Uses `triggeredAt` and `id` as a composite key for cursor-based pagination, which ensures stable pagination while maintaining the semantic order of notifications.

Preserves existing order by clauses for relay style pagination and only adds rowId if it doesn't exist.

valentinyanakiev committed:
Improves pagination for notifications

Enhances pagination logic for in-app notifications to ensure correct ordering and stability, especially when using cursor-based pagination.

Uses `triggeredAt` and `id` as a composite key for cursor-based pagination, which ensures stable pagination while maintaining the semantic order of notifications.

Preserves existing order by clauses for relay style pagination and only adds rowId if it doesn't exist.

hero101 commented on src/core/pagination/relay.style.pagination.fn.ts:
```diff
@@ -84,10 +84,22 @@ export const getRelayStylePaginationResults = async <
-    [`${query.alias}.${SORTING_COLUMN}`]: sort,
-    ...query.expressionMap.orderBys,
-  });
+  // Preserve existing orderBy clauses and only add rowId if not already present
```
`rowId` is always present.
What do we win with this new code?

hero101 commented on src/platform/in-app-notification/in.app.notification.service.ts:
```diff
@@ -107,9 +107,85 @@ export class InAppNotificationService {
+      );
+    } else {
+      // For simple pagination (first page), we can use the standard approach
+      queryBuilder.addOrderBy('notification.rowId', 'DESC');
```
the sort order is already defined by `getRelayStylePaginationResults`

hero101 reviewed (changes requested):
I have tested it and it works as expected.
I am unsure about of the changes in the core logic.

valentinyanakiev committed:
Improves in-app notification pagination

Removes unnecessary ordering for simple pagination to leverage the default ordering behavior. This improves efficiency when retrieving the first page of notifications.
